src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx

```diff
@@ -21,22 +25,37 @@ const ContactTagAddTagText = styled(Typography)(({ theme }) => ({
 }));
 
 interface ContactTagsProps {
+  accountListId: string;
   contactId: string;
-  contactTags: string[];
 }
 
 export const ContactTags: React.FC<ContactTagsProps> = ({
+  accountListId,
   contactId,
-  contactTags,
 }) => {
   const { t } = useTranslation();
+  const { data, loading } = useGetContactTagsQuery({
+    variables: {
+      accountListID: accountListId, // value for 'accountListID'
+      contactId: contactId, // value for 'contactId'
+    },
+  });
+
   const handleTagDelete = (tag: string) => {
-    const index = contactTags.indexOf(tag);
-    if (index > -1) {
-      contactTags = contactTags.splice(index, 1);
-      // TODO: add mutation to update tag for contact
-    }
-    console.info('You clicked the Delete tag for tag' + tag + '' + contactId);
+    console.info('You clicked the Delete tag for tag ' + tag + ' ' + contactId);
+    const list = data.contact.tagList;
+    const updatedTags: string[] = [];
+    list.forEach((element) => {
+      element != tag ? updatedTags.push(tag) : null;
+    });
+
+    const [] = useUpdateContactTagsMutation({
+      variables: {
+        accountList: accountListId,
+        contactId: contactId,
+        tagList: updatedTags,
+      },
+    });
```

Comment:
Hooks need to be at the top level of a react component. You can't put them inside other functions. This hook returns a function that you can use to call it when desired.
I would use the hook like this:
```ts
const [updateTags, { data }] = useUpdateContactTagsMutation({
  variables: {
    accountList: accountListId,
    contactId: contactId,
  },
});
```
and then later call the `updateTags` function to execute the mutation when needed
```ts
updateTags({
  variables: {
    tagList: updatedTags,
  },
});
```
You can set variables either in the hook or when you call the mutation. In this case you don't know the tags until you're inside this function so you can wait to set that until it's executed.

https://www.apollographql.com/docs/react/data/mutations/#executing-a-mutation